private/buf/bufcheck/internal/bufcheckserver/internal/bufcheckserverhandle/lint.go

```diff
@@ -190,6 +197,1022 @@ func handleLintDirectorySamePackage(
 	return nil
 }
 
+// HandleLintEnumNoAllowAlias is a handle function.
+var HandleLintEnumNoAllowAlias = bufcheckserverutil.NewLintEnumRuleHandler(handleLintEnumNoAllowAlias)
+
+func handleLintEnumNoAllowAlias(
+	responseWriter bufcheckserverutil.ResponseWriter,
+	_ bufcheckserverutil.Request,
+	enum bufprotosource.Enum,
+) error {
+	if enum.AllowAlias() {
+		responseWriter.AddProtosourceAnnotation(
+			enum.Location(),
+			nil,
+			`Enum option "allow_alias" on enum %q must be false.`,
+			enum.Name(),
+		)
+	}
+	return nil
+}
+
+// HandleLintEnumPascalCase is a handle function.
+var HandleLintEnumPascalCase = bufcheckserverutil.NewLintEnumRuleHandler(handleLintEnumPascalCase)
+
+func handleLintEnumPascalCase(
+	responseWriter bufcheckserverutil.ResponseWriter,
+	_ bufcheckserverutil.Request,
+	enum bufprotosource.Enum,
+) error {
+	name := enum.Name()
+	expectedName := stringutil.ToPascalCase(name)
+	if name != expectedName {
+		responseWriter.AddProtosourceAnnotation(
+			enum.NameLocation(),
+			nil,
+			"Enum name %q should be PascalCase, such as %q.",
+			name,
+			expectedName,
+		)
+	}
+	return nil
+}
+
+// HandleLintEnumFirstValueZero is a handle function.
+var HandleLintEnumFirstValueZero = bufcheckserverutil.NewLintEnumRuleHandler(handleLintEnumFirstValueZero)
+
+func handleLintEnumFirstValueZero(
+	responseWriter bufcheckserverutil.ResponseWriter,
+	_ bufcheckserverutil.Request,
+	enum bufprotosource.Enum,
+) error {
+	if values := enum.Values(); len(values) > 0 {
+		if firstEnumValue := values[0]; firstEnumValue.Number() != 0 {
+			// proto3 compilation references the number
+			responseWriter.AddProtosourceAnnotation(
+				firstEnumValue.NumberLocation(),
```

Comment:
The [original implementation passes extra ignore locations](https://github.com/bufbuild/buf/blob/c11801c352ea5c693faa3b8f43ec886e1339d765/private/bufpkg/bufcheck/buflint/internal/buflintcheck/buflintcheck.go#L192), but here it does not because it relies on the caller/CLI to handle extra ignore locations correctly. 

Other rules that do this are, `EnumValue*`, `EnumZeroValueSuffix`, `Field*`, `OneofLowerSnakeCase`, `Protovalidate`, `RPC*`.

Reply:
On second thought, what if the location here is `nil`? Then the caller/CLI will not be able to figure out the associated paths.

Actually, the statement ^ is not even correct. If the location is `nil`, currently `AddProtosourceAnnotation` would just panic. Just thinking out loud here.